src/NexusMods.DataModel/Games/RunGameTool.cs

```diff
@@ -107,6 +113,61 @@ public async Task Execute(Loadout loadout, ApplyPlan applyPlan, CancellationToke
         _logger.LogInformation("Finished running {Program}", program);
     }
 
+    private async Task RunThroughSteam(uint appId, CancellationToken cancellationToken)
+    {
+        await _osInterop.OpenUrl(new Uri($"steam://rungameid//{appId.ToString(CultureInfo.InvariantCulture)}"), cancellationToken);
+
+        if (OSInformation.Shared.IsWindows)
+        {
+            // TODO:
+        } else if (OSInformation.Shared.IsLinux)
+        {
+            var steam = await WaitForProcessToStart("steam", TimeSpan.FromMinutes(1), cancellationToken);
+            if (steam is null) return;
+
+            // NOTE(erri120): Reaper is a custom tool for cleaning up child processes
+            // See https://github.com/sonic2kk/steamtinkerlaunch/wiki/Steam-Reaper for details.
+            var reaper = await WaitForProcessToStart("reaper", TimeSpan.FromMinutes(1), cancellationToken);
+            if (reaper is null) return;
+
+            await reaper.WaitForExitAsync(cancellationToken);
+        }
+        else
+        {
+            throw OSInformation.Shared.CreatePlatformNotSupportedException();
+        }
+    }
+
+    private async ValueTask<Process?> WaitForProcessToStart(
+        string processName,
+        TimeSpan timeout,
+        CancellationToken cancellationToken = default)
+    {
+        try
+        {
+            var start = DateTime.UtcNow;
+            while (!cancellationToken.IsCancellationRequested && start + timeout < DateTime.UtcNow)
+            {
+                var processes = Process.GetProcessesByName(processName);
+                var target = processes.FirstOrDefault();
+                if (target is not null) return target;
```

Comment:
Not sure why this requires LINQ when the returned value is an array, but it's not a big deal.